### PR TITLE
refactor(forms): replace type any for the providers

### PR DESCRIPTION
**Pull request by JeanMeche**
The providers for the directives in forms can be typed as StaticProvider. Also the export is not required.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

File: packages/forms/src/directives/checkbox_value_accessor.ts [hunk 1]
```diff
@@ -6,11 +6,11 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive, ElementRef, forwardRef, Renderer2} from '@angular/core';
+import {Directive, forwardRef, StaticProvider} from '@angular/core';
 
 import {BuiltInControlValueAccessor, ControlValueAccessor, NG_VALUE_ACCESSOR} from './control_value_accessor';
 
-export const CHECKBOX_VALUE_ACCESSOR: any = {
+const CHECKBOX_VALUE_ACCESSOR: StaticProvider = {
```
Comment (AndrewKushnir) [inline packages/forms/src/directives/checkbox_value_accessor.ts:13]: nit: can we use the `Provider` type instead?

Comment (JeanMeche) [reply packages/forms/src/directives/checkbox_value_accessor.ts:13]: For my knowledge, why would we prefer `Provider` instead of `StaticProvider` here ?

**Review by AndrewKushnir — APPROVED** [review]
@JeanMeche thanks for the PR, the change looks great 👍

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 4dcbb6aef9ec6d1f1fe9a926d0b40c72139a013b.

**Pull request closed without merging**